src/coreclr/tools/SuperFileCheck/Program.cs

```diff
@@ -0,0 +1,623 @@
+                {
+                    return result;
+                }
+
+                return lineStr;
+            }
+        }
+
+        static string TransformMethod(MethodDeclarationSyntax methodDecl, string[] checkPrefixes)
+        {
+            return String.Join(Environment.NewLine, methodDecl.GetText().Lines.Select(x => TransformLine(x, checkPrefixes)));
+        }
+
+        static int GetMethodStartingLineNumber(MethodDeclarationSyntax methodDecl)
+        {
+            var leadingTrivia = methodDecl.GetLeadingTrivia();
+            if (leadingTrivia.Count == 0)
+            {
+                return methodDecl.GetLocation().GetLineSpan().StartLinePosition.Line;
+            }
+            else
+            {
+                return leadingTrivia[0].GetLocation().GetLineSpan().StartLinePosition.Line;
+            }
+        }
+
+        static string PreProcessMethod(MethodDeclarationInfo methodDeclInfo, string[] checkPrefixes)
+        {
+            var methodDecl = methodDeclInfo.Syntax;
+            var methodName = methodDeclInfo.Name;
+
+            // Create anchors from the first prefix.
+            var startAnchorText = $"// {checkPrefixes[0]}-LABEL: {methodName}(";
+            var endAnchorText = $"// {checkPrefixes[0]}: {methodName}(";
+
+            // Create temp source file based on the source text of the method.
+            // Newlines are added to pad the text so FileCheck's error messages will correspond
+            // to the correct line and column of the original source file.
+            // This is not perfect but will work for most cases.
+            var lineNumber = GetMethodStartingLineNumber(methodDecl);
+            var tmpSrc = new StringBuilder();
+            for (var i = 1; i < lineNumber; i++)
+            {
+                tmpSrc.AppendLine(String.Empty);
+            }
+            tmpSrc.AppendLine(startAnchorText);
+            tmpSrc.AppendLine(TransformMethod(methodDecl, checkPrefixes));
+            tmpSrc.AppendLine(endAnchorText);
+
+            return tmpSrc.ToString();
+        }
+
+
+        static async Task<FileCheckResult> RunSuperFileCheckAsync(MethodDeclarationInfo methodDeclInfo, string[] args, string[] checkPrefixes, string tmpFilePath)
+        {
+            File.WriteAllText(tmpFilePath, PreProcessMethod(methodDeclInfo, checkPrefixes));
+
+            try
+            {
+                args[0] = tmpFilePath;
+                return await RunLLVMFileCheckAsync(args);
+            }
+            finally
+            {
+                try { File.Delete(tmpFilePath); } catch { }
+            }
+        }
+
+        static bool IsArgumentCSharp(string arg)
+        {
+            return arg.Equals(CommandLineArgumentCSharp);
+        }
+
+        static bool IsArgumentCSharpListMethodNames(string arg)
+        {
+            return arg.Equals(CommandLineArgumentCSharpListMethodNames);
+        }
+        static bool IsArgumentCSharpFile(string arg)
```

Comment:
I could possibly be convinced otherwise, but I don't think I like the checking of the file extension.  If the user says it's a C# file, shouldn't we just use it?

Reply:
That's fair considering there are other possible extensions like `.csx`.